### PR TITLE
Closes #3262: Modify unnecessarily exported methods to unexported

### LINKED ISSUE #3262: Cleanup: multiple unnecessarily exported methods
Repurposing a script used to find unused test builder methods from https://github.com/tektoncd/pipeline/issues/3178#issuecomment-695158605 to try to find exported method names that aren't invoked from outside their package.

```
$ git grep -o "^func [A-Z][^(]*" pkg/ | grep -v "pkg/client" | grep -v " Test" | cut -d' '  -f 2 | sort | uniq | xargs -n 1 -I{} sh -c "git grep -c '\.{}(' > /dev/null || echo {}"
ApplyResourceSubstitution
ConvertErrorf
DagFromState
EmitErrorEvent
EmitEvent
EventForObjectWithCondition
EventForPipelineRun
EventForTaskRun
GetPVCSpec
GetPersistentVolumeClaim
HasDefaultConfigurationName
IsPodHitConfigError
MarkStatusFailure
MarkStatusRunning
MarkStatusSuccess
NeedsPVC
NewArtifactBucketFromConfig
NewFakeClient
NewPipelineRunTest
NewTektonCloudEventData
ResolvePipelineTaskResources
ShowRef
SubmoduleFetch
ValidateDeclaredWorkspaces
ValidateResults
WithDefaultConfigurationName
```

This greps for exported function names like `Foo` which aren't found in the codebase `.Foo(`, which would indicate that it's called by another package -- callers in the same package wouldn't need to reference the package name with a `.`

This also:
- ignores funcs in `pkg/client` since those are generated
- ignores funcs with `Test` in the name, since those are test methods called by `go test`

Spot-checking some of these:
- `SubmoduleFetch` is exported in `pkg/git/git.go` and only ever called in that package. It could be unexported.
- `ValidateDeclaredWorkspaces`, same story.
- `NeedsPVC` is exported and only called in its corresponding test, it could be unexported.
- `DagFromState` is defined in a `_test.go` file and only called from other tests, it could be unexported. (If it stays exported it should also be renamed to `DAGFromState` to adhere to Go style advice for [initialisms](https://github.com/golang/go/wiki/CodeReviewComments#initialisms)

I have a strong suspicion we could rename all these methods to be unexported (e.g., `ValidateResults` becomes `validateResults`) and check that tests still pass and images still build.

-----

Exported methods present a potential maintenance burden on Tekton contributors, since some external user might come to depend on the method, and could be broken if we change or remove the method.

Exported methods are effectively part of our Go library API, and while we don't have any support/deprecation policy around that, that wouldn't stop users from depending on it and being upset when we change it.

In general, methods should not be exported unless they need to be.

/kind cleanup

### PR DESCRIPTION
**Pull request by kobayashi**
# Changes

This is for issue #3262.

Only `ConvertErrorf` is kept as it is in v1beta1 because the function is imported in v1alpha1.

```
$ grep -r ConvertErrorf *pkg/apis/pipeline/v1alpha1/conversion_error.go:39:var convertErrorf = v1beta1.ConvertErrorf
pkg/apis/pipeline/v1beta1/conversion_error.go:45:// ConvertErrorf creates a CannotConvertError from the field name and format string.
pkg/apis/pipeline/v1beta1/conversion_error.go:46:func ConvertErrorf(field, msg string, args ...interface{}) error {
$
```

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [ ] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [ ] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)
- [ ] Release notes block has been filled in or deleted (only if no user facing changes)

_See [the contribution guide](https://github.com/tektoncd/pipeline/blob/master/CONTRIBUTING.md) for more details._

Double check this list of stuff that's easy to miss:

- If you are adding [a new binary/image to the `cmd` dir](../cmd), please update
  [the release Task](../tekton/publish.yaml) to build and release this image.

## Reviewer Notes

If [API changes](https://github.com/tektoncd/pipeline/blob/master/api_compatibility_policy.md) are included, [additive changes](https://github.com/tektoncd/pipeline/blob/master/api_compatibility_policy.md#additive-changes) must be approved by at least two [OWNERS](https://github.com/tektoncd/pipeline/blob/master/OWNERS) and [backwards incompatible changes](https://github.com/tektoncd/pipeline/blob/master/api_compatibility_policy.md#backwards-incompatible-changes) must be approved by [more than 50% of the OWNERS](https://github.com/tektoncd/pipeline/blob/master/OWNERS), and they must first be added [in a backwards compatible way](https://github.com/tektoncd/pipeline/blob/master/api_compatibility_policy.md#backwards-compatible-changes-first).

# Release Notes

### CODE REVIEW

File: pkg/contexts/contexts.go
Comment (vdemeester) [inline pkg/contexts/contexts.go:36]: If those are not exported, this means they are not used. We could remove them entirely :wink:

**Review by vdemeester — COMMENTED** [review]
Overall seems ok. I am a bit worried on `pkg/git` being used elsewhere (as a package) but it should be fine.

/ok-to-test

File: pkg/contexts/contexts.go
Comment (kobayashi) [reply pkg/contexts/contexts.go]: Thank you for the review. I removed these functions, also removed `emitEvent` `emitErrorEvent` from [pkg/reconciler/events/event.go](https://github.com/tektoncd/pipeline/pull/3289/files#diff-abd2046939930e89761ba190f55b5500) which are unused.

File: pkg/reconciler/events/event.go
Comment (afrittoli) [inline pkg/reconciler/events/event.go:106]: These were added because of backward compatibility,  but since we're cleaning up all unnecessarily exported methods, I'd say it makes sense to remove them.

**Review by afrittoli — COMMENTED** [review]
Thanks for this!
/lgtm

**Review by vdemeester — COMMENTED** [review]
/lgtm
/cc @ImJasonH

**Comment by kobayashi** [comment]
Just rebased to the latest. Please let me know anything I can do to proceed this PR.

**Review by vdemeester — COMMENTED** [review]
/lgtm

**Review by sbwsg — APPROVED** [review]

**Comment by kobayashi** [comment]
Hi, I just make sure if I still need some changes.

**Comment by ImJasonH** [comment]
> Hi, I just make sure if I still need some changes.

The PR needs to be rebased on `master`, but other than that the change still lgtm, so we should be able to get it merged after you rebase.

**Comment by kobayashi** [comment]
Thank you! 
Just rebased my PR repo.

**Comment by kobayashi** [comment]
@ImJasonH , could you review this again?

**Comment by ImJasonH** [comment]
/lgtm

Sorry for the delay, and thanks for doing this!

**Comment by vdemeester** [comment]
/lgtm cancel
Needs a rebase, cancelling the lgtm to unblock the merge queue 🙏🏼

**Comment by vdemeester** [comment]
(interesting.. it doesn't show anymore..)
/lgtm

**Comment by afrittoli** [comment]
I'm dropping lgmt to unlock tide as this needs a rebase

**Comment by vdemeester** [comment]
@afrittoli interesting.. when in the merge loop (all checks ok), it shows the rebase message from github, when not, it doesn't show it anymore.. 😓  Putting a hold for now while we wait for a rebase.
/hold

@kobayashi can you rebase your PR against master ?

**Comment by kobayashi** [comment]
Thank you, @vdemeester .
Rebased it again to master.

**Pull request merged**
